File: update_i18n.js
const fs = require('fs');
const path = require('path');
const dir = path.join(process.cwd(), 'src/screens');
const files = fs.readdirSync(dir).filter(f => f.endsWith('.tsx'));

for (const file of files) {
  const filePath = path.join(dir, file);
  let content = fs.readFileSync(filePath, 'utf8');
  
  let updated = false;
  
  if (content.includes("t('onboarding.")) {
    content = content.replace(/t\('onboarding\./g, "t('");
    updated = true;
  }
  
  if (file !== 'PhoneLoginScreen.tsx' && file !== 'OTPVerificationScreen.tsx') {
    if (content.includes("useTranslation(['common'])")) {
      content = content.replace(/useTranslation\(\['common'\]\)/g, "useTranslation(['onboarding'])");
      updated = true;
    }
  }
  
  if (updated) {
    fs.writeFileSync(filePath, content);
    console.log('Updated', file);
  }
}

File: update_i18n.py
import os
import re
import json

src_dir = 'src'
i18n_path = 'src/i18n/locales/en/common.json'

with open(i18n_path, 'r', encoding='utf-8') as f:
    en_json = json.load(f)

# Regex to find t('key', 'default')
pattern = re.compile(r"t\(\s*['\"]([^'\"]+)['\"]\s*,\s*['\"]([^'\"]+)['\"]\s*\)")
# Also find t('key') maybe? The user specifically said t('a11yXxx', ...) with fallback.

new_keys = 0
for root, _, files in os.walk(src_dir):
    for file in files:
        if file.endswith(('.ts', '.tsx')):
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
                matches = pattern.findall(content)
                for key, default_val in matches:
                    if key.startswith('a11y') and key not in en_json:
                        en_json[key] = default_val
                        new_keys += 1

with open(i18n_path, 'w', encoding='utf-8') as f:
    json.dump(en_json, f, indent=2, ensure_ascii=False)

print(f'Added {new_keys} new keys to common.json')
